content/courses/Ada_For_The_C_Embedded_Developer/chapters/05_SPARK.rst

```diff
@@ -7,36 +7,105 @@ Enhancing Verification with SPARK and Ada
+stack entry when called, for example a procedure or function body. There
+are a few other constructs that are also pertinent but this definition
+will suffice for now.
+
+Frames can have "exception handlers" that specify the response to be
+executed when an exception is "raised." These handlers can be specified to
+"handle" one or more specific exceptions, thus allowing the response to
+be tailored to specific exceptions or generalized to more than one.
+
+Structurally, the exception handlers are grouped together and textually
+separated from the rest of the code in a frame. As a result, there is a
+sequence of statements representing the normal flow of execution,
+distinct from another section representing the error handling. The
+reserved word "exception" separates these two sections. This separation
+helps simplify the overall flow, increasing understandability. In
+particular, status result codes are not required so there is no mixing
+of error checking and normal progress. If no exception is raised the
+exception handler section is automatically skipped during execution.
+
+For example, in the following procedure line 4 separates the two
+sections. Lines 5 through 10 contain the error handling part,
+consisting in this case of three exception handlers:
+
+:code-config:`accumulate_code=True`
+
+.. code:: ada project=Courses.Ada_For_C_Embedded_Dev.SPARK.Exceptions
+
+    procedure P is
+    begin
+       Statements_That_Might_Raise_Exceptions;
+    exception
+       when Exception_A =>
+          Handle_Exception_A;
+       when Exception_B =>
+          Handle_Exception_B;
+       when Exception_C =>
+          Handle_Exception_C;
+    end P;
+
+The name :ada:`Exception_A` refers to a specific visible exception
+declared somewhere with that name, as do the other two exceptions
+referenced. Each handler consists of an arbitrary sequence of
+statements, in this case specific procedures called in response to those
+specific exceptions.
+
+Note how the syntactic structure of the exception handling section
+resembles that of a case statement. The resemblance is intentional, to
+suggest similar behavior. When something in the statements of the normal
+execution raises an exception, normal execution is abandoned and the
+corresponding exception handler for that specific exception is executed.
+Only that one handler is executed and then the routine completes. The
+handlers do not "fall through" to the handlers below, again like the
+behavior of case statements.
+
+The semantic model is the same as with many other programming languages,
+in which normal execution is unavoidably interrupted in the face of an
+active exception. This design precludes missing (intentionally or
+otherwise) a function status result indicating an error, thus giving the
+developer a strong guarantee that errors are detected and addressed.
+
+When no corresponding handler is located in the exception handlers 
```

Comment:
Editorial suggestion: I'm not sure I'd understand this paragraph if I didn't know how exceptions are supposed to work. Maybe the whole paragraph could be merged with the paragraph below that explains the source-code example.